### PR TITLE
added missing dependency for py-msgpack that breaks neovim

### PR DESCRIPTION
**Pull request by robertu94**

### CODE REVIEW

**Review by adamjstewart — CHANGES_REQUESTED** [review]
What's the error message you see? msgpack has no imports of greenlet anywhere...

**Comment by robertu94** [comment]
If you build neovim, and try to use py-nvim-remote or py-nvim you get an import error for greenlet when you call ':checkhealth'

**Comment by adamjstewart** [comment]
What is the stacktrace of that import error? It doesn't seem to be coming from msgpack...

**Comment by robertu94** [comment]
@adamjstewart good catch.  Sorry for the noise.  The change should have been on py-pynvim.

Here is where the requirement is declared: https://github.com/neovim/pynvim/blob/d549371b6fb647304cff5f88a71e2bc42d2ed30a/setup.py#L33

**Comment by robertu94** [comment]
The stack trace looks like this:

```console
## Python 3 provider (optional)                                                                         
  - INFO: `g:python3_host_prog` is not set.  Searching for python3 in the environment.                  
  - INFO: Multiple python3 executables found.  Set `g:python3_host_prog` to avoid surprises.            
  - INFO: Executable: /gpfs/fs1/home/runderwood/git/spack-bebop/opt/spack/linux-centos7-broadwell/gcc-    10.2.0/python-3.9.13-atmnl5spo3qj7nsu44fbhlefwjrczmtr/bin/python3                                     
  - INFO: Other python executable: /usr/bin/python3                                                     
  - ERROR: Command error (job=7, exit code 1): `'/gpfs/fs1/home/runderwood/git/spack-bebop/opt/spack/     linux-centos7-broadwell/gcc-10.2.0/python-3.9.13-atmnl5spo3qj7nsu44fbhlefwjrczmtr/bin/python3' -c       'import sys; sys.path = list(filter(lambda x: x != "", sys.path)); import neovim; print(neovim.         __file__)'` (in '/gpfs/fs1/home/runderwood/git/spack-bebop')
    stderr: Traceback (most recent call last):  File "<string>", line 1, in <module>  File "/gpfs/fs1/      home/runderwood/git/spack-bebop/opt/spack/linux-centos7-broadwell/gcc-10.2.0/py-pynvim-0.4.3-           nuu2k6amlhgdfuqp575ifdwbscjwjtbw/lib/python3.9/site-packages/neovim/__init__.py", line 5, in            <module>    import pynvim  File "/gpfs/fs1/home/runderwood/git/spack-bebop/opt/spack/linux-centos7-     broadwell/gcc-10.2.0/py-pynvim-0.4.3-nuu2k6amlhgdfuqp575ifdwbscjwjtbw/lib/python3.9/site-packages/      pynvim/__init__.py", line 11, in <module>    from pynvim.msgpack_rpc import (ErrorResponse,             child_session, socket_session,  File "/gpfs/fs1/home/runderwood/git/spack-bebop/opt/spack/linux-        centos7-broadwell/gcc-10.2.0/py-pynvim-0.4.3-nuu2k6amlhgdfuqp575ifdwbscjwjtbw/lib/python3.9/site-       packages/pynvim/msgpack_rpc/__init__.py", line 10, in <module>    from pynvim.msgpack_rpc.session       import ErrorResponse, Session  File "/gpfs/fs1/home/runderwood/git/spack-bebop/opt/spack/linux-         centos7-broadwell/gcc-10.2.0/py-pynvim-0.4.3-nuu2k6amlhgdfuqp575ifdwbscjwjtbw/lib/python3.9/site-       packages/pynvim/msgpack_rpc/session.py", line 7, in <module>    import greenletModuleNotFoundError:     No module named 'greenlet'
  - INFO: Python version: 3.9.13                                                                        
  - INFO: pynvim version: unable to load neovim Python module                                           
  - ERROR: pynvim is not installed.                                                                     
    Error: unable to load neovim Python module                                                              - ADVICE:                                                                                           
      - Run in shell: /gpfs/fs1/home/runderwood/git/spack-bebop/opt/spack/linux-centos7-broadwell/gcc-        10.2.0/python-3.9.13-atmnl5spo3qj7nsu44fbhlefwjrczmtr/bin/python3 -m pip install pynvim
```

File: var/spack/repos/builtin/packages/py-pynvim/package.py [hunk 1]
```diff
@@ -17,5 +17,6 @@ class PyPynvim(PythonPackage):
     version("0.4.3", sha256="3a795378bde5e8092fbeb3a1a99be9c613d2685542f1db0e5c6fd467eed56dff")
 
     depends_on("py-setuptools", type="build")
+    depends_on("py-greenlet", type=("build", "run"))
 
     depends_on("py-msgpack", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pynvim/package.py]:
```suggestion
    depends_on("py-msgpack@0.5:", type=("build", "run"))
```
There are also some missing dependencies for Python 3.3 and older, not sure if you want to add them:
```python
depends_on("py-pyuv@1:", when="^python@:3.3 platform=windows", type=("build", "run"))
for platform in ["linux", "darwin", "cray"]:
    depends_on("py-trollius", when="^python@:3.3 platform=" + platform, type=("build", "run"))

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**